frontend/templates/product-list/sections/FilterableProductsSection/RefinementSingleSelect.tsx

```diff
@@ -106,7 +89,21 @@ const SingleSelectItem = React.memo(function SingleSelectItem({
          as={shouldBeLink ? 'a' : 'button'}
          onClick={(event) => {
             event.preventDefault();
-            onClick && onClick(item);
+            setIndexUiState((prevUiState) => {
+               const refinementList = {
+                  ...prevUiState.refinementList,
+                  [attribute]: [item.value],
+               };
+               if (
+                  item.value === prevUiState?.refinementList?.[attribute]?.[0]
+               ) {
+                  delete refinementList[attribute];
+               }
```

Comment:
This is the only other way I can think of, but I think what you have is fine

```suggestion
               const refinementList = item.value === prevUiState?.refinementList?.[attribute]?.[0]
                  ? {
                        ...prevUiState.refinementList,
                        [attribute]: undefined,
                  } : {
                        ...prevUiState.refinementList,
                        [attribute]: [item.value],
                  };
```